python/ray/train/__init__.py

```diff
@@ -2,14 +2,29 @@
 from ray.train.backend import BackendConfig
 from ray.train.data_config import DataConfig
 from ray.train.constants import TRAIN_DATASET_KEY
+from ray.train._internal.session import (
+    get_dataset_shard,
+    get_trial_resources,
+    get_world_rank,
+    get_world_size,
+    report,
+)
 from ray.train.trainer import TrainingIterator
 
+from ray.air.config import RunConfig, ScalingConfig
 
 usage_lib.record_library_usage("train")
 
 __all__ = [
     "BackendConfig",
     "DataConfig",
+    "RunConfig",
+    "ScalingConfig",
     "TrainingIterator",
     "TRAIN_DATASET_KEY",
+    "get_dataset_shard",
+    "get_trial_resources",
```

Comment:
This is a Tune concept.

Reply:
We're moving this as well to Train right? You cannot launch trials without resources.

Reply:
Yes, this is a train concept and being used in train examples (e.g. to set the number of threads for trainers).

Reply:
Maybe we can discuss this along with:
> Make it so you can run any trainable func/class with Trainer.

Right now the interfaces are: 
`ScalingConfig` (Train) -> `PlacementGroupFactory` (Tune) -> `PlacementGroup` (Core)

I would say that the usage in Train examples is a misuse of the API and extremely confusing. For setting the number of threads, this is being used as a proxy when we actually just want an API to get the number of CPUs assigned to the `RayTrainWorker` actor.

Reply:
Yea the placement group stuff also needs to move in this case.